File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def helper(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        if root.left is None and root.right is None:
            return 1
        if root.left is None:
            return 1 + self.helper(root.right)
        if root.right is None:
            return 1 + self.helper(root.left)
        return 1 + min(self.helper(root.left), self.helper(root.right))
    def minDepth(self, root: Optional[TreeNode]) -> int:
        return self.helper(root)
        

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int helper(TreeNode* root){
        if (root == nullptr){
            return 0;
        }
        if (root->left == nullptr && root->right == nullptr){
            return 1;
        }
        if (root->left == nullptr){
            return 1 + helper(root->right);
        }
        if (root->right == nullptr){
            return 1 + helper(root->left);
        }
        return 1 + min(helper(root->left), helper(root->right));
    }
    int minDepth(TreeNode* root) {
        return helper(root);
    }
};